codeowners.go

```diff
@@ -79,8 +79,13 @@ func NewCodeowners(path string) (*Codeowners, error) {
 	if r == nil {
 		return nil, fmt.Errorf("No CODEOWNERS found in %s", path)
 	}
+	return NewCodeownersFromReader(r, root)
+}
+
+// NewCodeownersFromReader creates a Codeowners from a given Reader instance and root path.
+func NewCodeownersFromReader(r io.Reader, repoRoot string) (*Codeowners, error) {
```

Comment:
I regret the original `NewCodeowners` function name - calling `codeowners.NewCodeowners` is awkward... How about simplifying this name:

```suggestion
// FromReader creates a Codeowners from a given Reader instance and root path.
func FromReader(r io.Reader, repoRoot string) (*Codeowners, error) {
```

Reply:
Updated with some cleanup. Let me know what you think!